source/common/res/features/budget-daily-spending/main.js

```diff
@@ -0,0 +1,127 @@
+          let categoryName = ynabToolKit.dailySpending.getInspectorName();
+          let masterCategoryViewId = $('ul.is-checked').prevAll('ul.is-master-category').attr('id');
+          let masterCategory = ynabToolKit.shared.getEmberView(masterCategoryViewId).get('data');
+          let masterCategoryId = masterCategory.get('categoryId');
+
+          categories.forEach(function (f) {
+            if (f.name === categoryName && f.masterCategoryId === masterCategoryId) {
+              ynabToolKit.dailySpending.updateDailySpending(f);
+
+              return false;
+            }
+          });
+        },
+
+        observe(changedNodes) {
+          if (changedNodes.has('navlink-budget active') ||
+              changedNodes.has('budget-table-cell-available-div user-data') ||
+              changedNodes.has('budget-inspector')) {
+            ynabToolKit.dailySpending.invoke();
+          }
+        },
+
+        addBudgetVersionIdObserver() {
+          let applicationController = ynabToolKit.shared.containerLookup('controller:application');
+          applicationController.addObserver('budgetVersionId', function () {
+            Ember.run.scheduleOnce('afterRender', this, resetBudgetViewDailySpending);
+          });
+
+          function resetBudgetViewDailySpending() {
+            ynabToolKit.dailySpending.budgetView = null;
+          }
+        },
+
+        getCategories() {
+          if (ynabToolKit.dailySpending === 'undefined') {
+            return [];
+          }
+
+          // After using Budget Quick Switch, budgetView needs to be reset to the new budget.
+          if (ynabToolKit.dailySpending.budgetView === null) {
+            try {
+              ynabToolKit.dailySpending.budgetView = ynab.YNABSharedLib.getBudgetViewModel_AllBudgetMonthsViewModel()._result;
+            } catch (e) {
+              return;
+            }
+          }
+          var categories = [];
+          var masterCats = ynabToolKit.dailySpending.budgetView
+          .categoriesViewModel.masterCategoriesCollection._internalDataArray;
+          var masterCategories = [];
+
+          masterCats.forEach(function (c) {
+            // Filter out "special" categories
+            if (c.internalName === null) {
+              masterCategories.push(c.entityId);
+            }
+          });
+
+          masterCategories.forEach(function (c) {
+            var accounts = ynabToolKit.dailySpending.budgetView
+              .categoriesViewModel.subCategoriesCollection
+              .findItemsByMasterCategoryId(c);
+
+            Array.prototype.push.apply(categories, accounts);
+          });
+
+          return categories;
+        },
+
+        updateDailySpending(f) {
+          var amount = ynabToolKit.dailySpending.getBalanceAmount(f);
+          var date = new Date();
+          var remainingDays = new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate() - date.getDate();
+          var dailySpendingAmount = amount / remainingDays;
```

Comment:
I only ask this because even though it's simple math, when the math is dense, it's often hard to debug later.